src/Microsoft.Extensions.Http.Polly/HttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Http;
+using System;
+using System.Net.Http;
+
+namespace Polly
+{
+    /// <summary>
+    /// Extension methods for <see cref="HttpRequestMessage"/> Polly integration.
+    /// </summary>
+    public static class HttpRequestMessageExtensions
+    {
+        internal static readonly string PollyContextKey = "PollyContext";
+        
+        /// <summary>
+        /// Gets the <see cref="Context"/> associated with the provided <see cref="HttpRequestMessage"/>.
+        /// </summary>
+        /// <param name="request">The <see cref="HttpRequestMessage"/>.</param>
+        /// <returns>The <see cref="Context"/> is set, otherwise <c>null</c>.</returns>
+        /// <remarks>
+        /// The <see cref="PolicyHttpMessageHandler"/> will attach a context to the <see cref="HttpResponseMessage"/> prior
+        /// to executing a <see cref="Policy"/>, if one does not already exist. The <see cref="Context"/> will be provided
+        /// to the policy for use inside the <see cref="Policy"/> and in other message handlers.
+        /// </remarks>
+        public static Context GetPollyContext(this HttpRequestMessage request)
```

Comment:
Thoughts on naming? I also considered `GetPolicyContext`, which I kinda like better on second thought.

Reply:
We could also give this a scarier name like `GetPolicyExecutionContext` to act as salt.

Reply:
My vote on naming is for `GetPolicyExecutionContext()`: reinforces the idea that the context is execution-scoped.  If so, should the change flow elsewhere also?, eg:

    internal static readonly string PolicyExecutionContextKey = "PolicyExecutionContext";

Reply:
Sure, will make this change.